docs/BR.md

```diff
@@ -994,25 +1026,88 @@ Databases maintained by the CA, its owner, or its affiliated companies do not qu
+
+A Network Perspective MAY use a recursive DNS resolver that is NOT co-located with the Network Perspective. However, the DNS resolver used by the Network Perspective MUST fall within the same Regional Internet Registry service region as the Network Perspective relying upon it. Furthermore, for any pair of DNS resolvers used on a Multi-Perspective Issuance Corroboration attempt, the straight-line distance between the two States, Provinces, or Countries the DNS resolvers reside in MUST be at least 500 km. The location of a DNS resolver is determined by the point where unencapsulated outbound DNS queries are first handed off to the network infrastructure providing Internet connectivity to that DNS resolver.
+
+CAs MAY immediately retry Multi-Perspective Issuance Corroboration using the same validation method or an alternative method (e.g., a CA can immediately retry validation using "Email to DNS TXT Contact" if "Agreed-Upon Change to Website - ACME" does not corroborate the outcome of Multi-Perspective Issuance Corroboration). When retrying Multi-Perspective Issuance Corroboration, CAs MUST NOT rely on corroborations from previous attempts. There is no stipulation regarding the maximum number of validation attempts that may be performed in any period of time.
+
+The "Quorum Requirements" Table describes quorum requirements related to Multi-Perspective Issuance Corroboration. Network Perspectives are considered distinct when the straight-line distance between the two States, Provinces, or Countries they reside in is at least 500 km. Network Perspectives are considered "remote" when they are distinct from the Primary Network Perspective and the other Network Perspectives represented in a quorum.
+
+A CA MAY reuse corroborating evidence for CAA record quorum compliance for a maximum of 398 days. After issuing a Certificate to a domain, remote Network Perspectives MAY omit retrieving and processing CAA records for its subdomains in subsequent Certificate requests from the same Applicant for up to a maximum of 398 days.
+
+Table: Quorum Requirements
+
+| # of Distinct Remote Network Perspectives Used | # of Allowed non-Corroborations |
+| --- | --- |
+| 2-5 |  1  |
+| 6+  |  2  |
+
+Remote Network Perspectives performing Multi-Perspective Issuance Corroboration:
```

Comment:
I agree with the spirit of this requirement, but I am concerned that it isn't entirely feasible. I recently spoke to a team that deploys RPKI-invalid route filtering and they mentioned that they may not enable it for certain types of peers. They will also disable it if there is an issue with the filtering system itself or they may grant a temporary exception for a customer. I would assume this is common?

As I currently read the text, the CA would be required to file an incident and halt issuance if any of these exceptions were made. CAs are very unlikely to have an SLA with their network operator to always support RPKI-invalid route filtering. 

How should this situation be handled ideally?

Reply:
@jdkasten - thanks for you comment!

Is it safe to assume the specific comment is focused on the following bullet?:

- [Remote Network Perspectives performing Multi-Perspective Issuance Corroboration MUST]:  _Forward all Internet traffic via a network or set of networks that filter all RPKI-invalid BGP routes as defined by RFC 6811_"

**Proposed Update:** 

If so, what if instead the language was modified to something like the following:

- [Remote Network Perspectives performing Multi-Perspective Issuance Corroboration MUST]: _**under normal operating conditions**, forward all Internet traffic via a network or set of networks that filter **all or some** RPKI-invalid BGP routes as defined by RFC 6811._

Feedback on the above is welcome. 

**Change Justification:** 

As I understand it, there are two distinct categories of exceptions to RPKI filtering: 
- the first is temporary exceptions where certain peers are identified as potentially having an issue with RPKI ROV enforcement and these sessions are given a temporary exception. 
- the second is that some types of sessions are considered exempt from a network's filtering policy. There are documented cases of networks that filter routes from peers but not from customers (see [https://isbgpsafeyet.com/](https://isbgpsafeyet.com/).

The approach proposed above would consider both types of exceptions described above, while also addressing your concern re: risk of blocking issuance when filtering is beyond the CA Owner's control. 

FWIW, it would seem there are some tools available that would help CA owners evaluate whether ISPs are performing filtering, for example https://rpkitest.nlnetlabs.net/ from https://nlnetlabs.nl/research/projects/

Reply:
I'm not a network specialist.. how does one meet this MUST requirement?

Reply:
Great, thanks for confirming! 

We'll:
1) stage these changes in https://github.com/ChristopherRC/servercert/blob/SC-067-Version-2/docs/BR.md in preparation for the next round of discussion, and
2) leave this comment thread open in the event there's any other discussion.

Reply:
@romanf - the expectation is that the lookups performed by the remote network perspectives used for MPIC would rely on an ISP that performs RPKI-invalid route filtering.

Some examples are located at: https://isbgpsafeyet.com/ (note: I cannot speak to the completeness of this list, however, I would expect any ISP to be able to describe whether filtering is in place).

Reply:
So the CA would need written confirmation from all of their ISPs that they're using for the various perspectives that they perform RPKI-invalid route filtering. Correct?

Reply:
@romanf - I think there are a few ways CA Owners could collect evidence to demonstrate a service provider performs invalid-route filtering. Written confirmation from the service provider(s) seems like one way this could be addressed. Tools like https://rpkitest.nlnetlabs.net/ could also be used by CA Owners and their auditors to corroborate statements from service providers or lists like we see on https://isbgpsafeyet.com/.

Reply:
The changes proposed by @jdkasten are significant improvements! Having said that, Google cloud apparently does not yet meet the RPKI filtering requirement. There is currently a lot of active RPKI deployment activity, and I wonder if we should phase in this requirement by starting with a SHOULD and perhaps some language that forces CAs to think about how they can meet the requirement?

Reply:
@wthayer 

While we prefer the MUST due to the demonstrated security benefits of invalid route filtering, we understand your concern and we have proposed a phased approach at https://github.com/ChristopherRC/servercert/commit/d6125463d370f57a78d29c072c49c88dda7d579d. Feedback is welcome! The effective date for invalid route filtering in this proposal aligns with when we would expect to see blocking behavior for MPIC (September 15, 2025). Of course, CAs are encouraged to pursue earlier adoption.

Reply:
Thank you @ChristopherRC. I am particularly interested in having the ability to run perspectives on Google cloud, so I am trying to get some timing from them to determine if this change helps. In general it might be better to include something like this right above your proposed change:

Under normal operating conditions, remote Network Perspectives performing Multi-Perspective Issuance Corroboration SHOULD forward all Internet traffic via a network or set of networks that filter all or some RPKI-invalid BGP routes as defined by RFC 6811. The CA MUST document the reason(s) for all non-compliance.

Maybe this is going too far, but my intent is to force CAs to think about meeting this before the requirement actually comes into effect. Also, I'd want to improve that paragraph to remove ambiguity around the meaning of "non-compliance" before incorporating it, should you like to idea.

Reply:
Hi Folks - I work in Fastly's network engineering department and am available as RPKI and BGP Routing Security subject matter expert.

Reviewing the draft BR requirements from a network engineering perspective, to me it seems RFC 6811 might not be the best reference. RFC 6811 merely is an algorithm description. The outcome of the algorithm depends on the algorithm's inputs, which are the RPKI Trust Anchor Locators the network operator has chosen to use. For example, if zero RPKI Trust Anchors are used, the outcome of the RFC 6811 validation algorithm will always be `not-found`. Would a network applying RFC 6811 _without any Trust Anchors_ be compliant with the spirit of the requirement? I don't think so! :-)

However, explicitly prescribing which RPKI Trust Anchors a CA's network must/should use also has its challenges: Some Regional Internet Registries (RIRs) employ a Relying Party Agreement (RPA) to restrict access to their online Resource Certification PKI services. The CA/B BR mandating that a particular RPKI Trust Anchor MUST be used, also means the CA's network operator MUST agree the current and any future versions of such RPKI Relying Party Agreements. Some wiggle room probably is needed to allow network operators to make independent choices.

There is a non-zero number of networks which use only 4 of the 5 RIR Trust Anchors due to legal concerns with the 5th RIR's RPA. The world's second most popular RPKI validator [rpki-client](https://man.openbsd.org/rpki-client#FILES) does not include all 5 RIR Trust Anchor TALs in its distribution. But, I'd consider network operations using 4 out of 5 RIR Trust Anchor Locators to be compliant with the spirit of this requirement.

I'd like to echo @jdkasten's comments about RPKI-filtering generally speaking not being a binary state of 'on' or 'off'. In this [NANOG](https://mailman.nanog.org/pipermail/nanog/2021-April/213346.html) posting I've outlined reasons why a network at any given time might not have RPKI enabled on all its routers. If RPKI-filtering is disabled on just 1 router in a global IP network, it might appear to outsiders the network is not doing any RPKI-filtering at all.

It is also good to keep in mind that RPKI-based filtering is not the **only** approach to improve routing security and attempt to mitigate routing incidents. For example, RFC 9234 specifies an approach using in-band BGP signalling to help avoid route leaks without RPKI. 

I'd like to propose text along these lines:

> Networks used by Remote Network Perspectives performing Multi-Perspective Issuance Corroboration MUST implement measures to mitigate BGP routing incidents in the global Internet routing system. Network operators are RECOMMENDED to use mechanisms based on Secure Inter-Domain Routing (RFC 6480), for example, BGP Prefix Origin Validation (RFC 6811), make use of other non-RPKI route-leak prevention mechanisms (such as RFC 9234), and apply best current practices described in BCP 194.

I understand the above is a bit of a departure from the original text, but I think the original text's hard requirements might not be entirely realistic and might distract from the ultimate goal: encourage networks to undertake efforts to improve routing security. I think it is important to encourage RNP's to make use of networks that make an effort to avoid routing incidents.

Reply:
Hi @job,

1) We are familiar with and very much appreciate your contributions in the routing security space. Thank you for reviewing the proposal, offering concrete suggestions for improvement, and volunteering your expertise to help amplify the positive impact this ballot can have on the Web PKI and user security!

2) On the proposed update:

> Networks used by Remote Network Perspectives performing Multi-Perspective Issuance Corroboration MUST implement measures to mitigate BGP routing incidents in the global Internet routing system. Network operators are RECOMMENDED to use mechanisms based on Secure Inter-Domain Routing (RFC 6480), for example, BGP Prefix Origin Validation (RFC 6811), make use of other non-RPKI route-leak prevention mechanisms (such as RFC 9234), and apply best current practices described in BCP 194.

One challenge with this approach is that it imposes requirements on service providers that are not obligated to adhere to the CA/Browser Forum policies. Do you think the following re-framing could achieve the same security outcome as your proposal?

> Network Perspectives performing Multi-Perspective Issuance Corroboration MUST rely upon networks that implement measures to mitigate BGP routing incidents in the global Internet routing system. In support of performing Multi-Perspective Issuance Corroboration, CAs SHOULD rely upon network operators that: 

> 1. use mechanisms based on Secure Inter-Domain Routing (RFC 6480), for example, BGP Prefix Origin Validation (RFC 6811), 
> 2. make use of other non-RPKI route-leak prevention mechanisms (such as RFC 9234), and 
> 3. apply best current practices described in BCP 194.

> While It is RECOMMENDED that under normal operating conditions Network Perspectives performing Multi-Perspective Issuance Corroboration forward all Internet traffic via a network or set of networks that filter all or some RPKI-invalid BGP routes as defined by RFC 6811, it is NOT REQUIRED.

This approach attempts to blend the elements of your proposal, while also aligning with the strong recommendation of relying on RPKI-invalid route filtering shared with us by the Princeton research team at F2F 57 and the corresponding [research](https://arxiv.org/pdf/2302.08000.pdf) that inspired this work.

Comments on this approach are welcome and appreciated!

Reply:
Thank you @ryancdickson for the kind words.

Your re-framed version seems an improvement!

I would suggest to remove the words `all or some`.

Reply:
To elaborate a bit:

The Princeton research team's observation that a proper issuance process execution has a dependency on proper network reachability seems on the nose to me. MPIC to me seems a fantastic mechanism to attempt to overcome various potential issues which may exist in the Internet's routing system substrate, I'm excited to see MPIC advance. Recognizing that proper network reachability is a key ingredient to the issuance process - regardless of MIPC - encouraging all forms of BGP routing hygiene seems very helpful.

Reply:
Sorry if this question is a bit stupid... I'm not a network engineer. :)

How would a CA implement the MUST requirement
"Network Perspectives performing Multi-Perspective Issuance Corroboration MUST rely upon networks that implement measures to mitigate BGP routing incidents in the global Internet routing system." ?

Would this mean - in laymans-terms - that the networks where the remote perspecitves are located (or - if VPN are allowed - the VPN endpoint is) must implement this BGP mitigations? Or does this apply to all networks the CA is using (incl. their main internet access network as well as transit networks to the remote perspectives)?

Reply:
@romanf - not a "stupid" question. If there's opportunity to clarify these requirements either through discussion or updates to the ballot, they are worth addressing.

I would consider this a more flexible option than what we originally proposed (i.e., "_[Remote Network Perspectives performing Multi-Perspective Issuance Corroboration MUST]: under normal operating conditions, forward all Internet traffic via a network or set of networks that filter all or some RPKI-invalid BGP routes as defined by RFC 6811."_). 

This should be considered more flexible because rather than only focusing on RPKI-invalid route filtering to improve routing security relied upon by the Network Perspectives, it allows consideration for other routing security technologies as described by @job in [this](https://github.com/cabforum/servercert/pull/487#discussion_r1566070938) comment. That is to say, RPKI-invalid route filtering should not be the only acceptable approach adopted by network operators/ISPs to improve routing security that's described in the ballot. 

It's my understanding that if a CA Owner ensured its Network Perspectives relied upon service providers that filter RPKI-invalid routes (like many of those described on isbgpsafeyet.com) - they'd satisfy this expectation. 

Feedback from @job and other routing security experts welcome (as I am definitely not one!).

Reply:
The list at isbgpsafeyet.com is awfully short... and just to double-check: this requirement is only for the remote perspectives, correct? Otherwise I can imagine some CAs getting into deep troubles if the DCs that host their primary perspective can't be connected to any ISP providing the required GBP filtering...